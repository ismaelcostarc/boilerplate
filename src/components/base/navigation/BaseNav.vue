<script setup lang="ts">
  const props = defineProps<{
    items: {
      name: string
      route: string
    }[]
    activeIndex: number
  }>()
</script>

<template>
  <nav>
    <ul class="flex justify-between">
      <li
        v-for="(item, index) in props.items"
        :key="item.name"
        class="border-r last:border-0 border-black border-solid w-full text-center"
        :class="`text-gray-${index !== activeIndex ? 400 : 900}`"
      >
        <router-link :to="item.route">
          {{ item.name }}
        </router-link>
      </li>
    </ul>
  </nav>
</template>
