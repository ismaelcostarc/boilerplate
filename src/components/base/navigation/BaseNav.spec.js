import { mount } from '@vue/test-utils'
import BaseNav from './BaseNav.vue'
import { describe, it, expect, beforeEach } from 'vitest'

describe('BaseNav', () => {
  let wrapper

  beforeEach(() => {
    wrapper = mount(BaseNav, {
      props: {
        items: [
          {
            name: 'first',
            route: '/first-route',
          },
          {
            name: 'second',
            route: '/second-route',
          },
        ],
      },
    })
  })

  it('should mount component', () => {
    expect(BaseNav).toBeTruthy()

    expect(wrapper.text()).toContain('first')
    expect(wrapper.text()).toContain('second')
  })

  it('should be rendered correctly with props', () => {
    expect(wrapper.text()).toMatchSnapshot()
  })

  it('should be rendered correctly without props', () => {
    const wrapper = mount(BaseNav)

    expect(wrapper.text()).toMatchSnapshot()
  })
})
