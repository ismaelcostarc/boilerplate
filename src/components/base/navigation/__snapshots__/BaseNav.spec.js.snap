// Vitest Snapshot v1

exports[`BaseNav > should be rendered correctly with props 1`] = `"firstsecond"`;

exports[`BaseNav > should be rendered correctly without props 1`] = `""`;

exports[`BaseNav > should be rendered correctly without slots 1`] = `""`;
